include/jemalloc/internal/prof_internal_externs.h

```diff
@@ -0,0 +1,14 @@
+#ifndef JEMALLOC_INTERNAL_PROF_INTERNAL_EXTERNS_H
```

Comment:
Maybe keep these still in prof_externs? It’s a bit strange to have an _internal_externs

Reply:
Got rid of `prof_internal_externs.h` by -
- Moved `prof_getpid()` and `prof_get_default_filename()` back to `prof_externs.h`. After all, they're quite general and may be externally needed.
- Moved the two `*_mutex_choose()` functions into `prof_data.c` and made them static. Also relocated the corresponding mutexes there but exposed them so that they can still be handled in `prof.c`.